src/meta/src/stream/scale.rs

```diff
@@ -358,16 +357,18 @@ where
         let mut no_shuffle_source_fragment_ids = HashSet::new();
         let mut no_shuffle_target_fragment_ids = HashSet::new();
 
-        // Index the downstream fragment
-        let mut downstream_fragment_id_map = HashMap::new();
+        let mut fragment_dispatcher_map = HashMap::new();
+
         for actor in actor_map.values() {
+            // Question(peng): will a fragment use multiple different dispatchers to connect to the
+            // same downstream fragment?
```

Comment:
I think we can ignore the cases of multiple edges between two fragments for now, which might be hard to resolve as described [here](https://risingwave-labs.notion.site/Multiple-Edges-in-the-Stream-Graph-f0ab595523d74823a39edaebdea2be16). If there does exist, then a huge number of changes need to introduced in the meta service. 🥵